app/views/appointment_summaries/_preview_form.html.erb

```diff
@@ -7,6 +7,7 @@
   <% end %>
 
   <%= f.submit local_assigns.fetch(:submit_button_text),
-    class: (%w(btn btn-lg) | Array(local_assigns[:submit_button_classes])) %>
+    class: (%w(btn btn-lg) | Array(local_assigns[:submit_button_classes])),
+    onclick: 'this.disabled = true; this.parentElement.submit();' %>
```

Comment:
Good in principle, but makes too many assumptions about the DOM. A markup change could easily break this and automated tests wouldn't catch it.